File: LeetCode/Sept-2023/896. Monotonic Array/isMonotonic.py
class Solution:
    def isMonotonic(self, nums):
        n = len(nums)
        inc = True
        desc = True

        for i in range(1, n):
            if nums[i] - nums[i - 1] < 0:
                inc = False
            if nums[i] - nums[i - 1] > 0:
                desc = False

        return inc or desc

File: LeetCode/Sept-2023/896. Monotonic Array/isMonotonic.cpp
class Solution {
public:
    bool isMonotonic(std::vector<int>& nums) {
        int n = nums.size();
        bool inc = true;
        bool desc = true;

        for (int i = 1; i < n; i++) {
            if (nums[i] - nums[i - 1] < 0) {
                inc = false;
            }
            if (nums[i] - nums[i - 1] > 0) {
                desc = false;
            }
        }

        return inc || desc;
    }
};
